applications/ParticleMechanicsApplication/python_scripts/particle_vtk_output_process.py

```diff
@@ -1,249 +1,104 @@
+    if not isinstance(settings, KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ParticleVtkOutputProcess(model, settings["Parameters"])
+
+class ParticleVtkOutputProcess(KratosMultiphysics.OutputProcess):
+
+    def __init__(self, model, settings):
+        super().__init__()
+
+        model_part_name = settings["model_part_name"].GetString()
+        self.model_part = model[model_part_name]
 
-        self.param = param
-        self.model_part = model_part
-        self.problem_name = param["model_part_name"].GetString()
+        self.TranslateLegacyVariablesAccordingToCurrentStandard(settings)
 
+        # Default settings can be found in "custom_io/particle_vtk_output.cpp"
+        self.vtk_io = KratosParticle.ParticleVtkOutput(self.model_part, settings)
+
+        if settings["save_output_files_in_folder"].GetBool():
+            if self.model_part.GetCommunicator().MyPID() == 0:
+                output_path = settings["output_path"].GetString()
+                if not self.model_part.ProcessInfo[KratosMultiphysics.IS_RESTARTED]:
+                    kratos_utils.DeleteDirectoryIfExisting(output_path)
+            self.model_part.GetCommunicator().GetDataCommunicator().Barrier()
+
+        self.output_interval = settings["output_interval"].GetDouble()
+        self.output_control = settings["output_control_type"].GetString()
         self.next_output = 0.0
 
-        self.coords_X = np.empty(0)
-        self.coords_Y = np.empty(0)
-        self.coords_Z = np.empty(0)
-        self.temp_results = np.empty([0,3])
-        self.result_names = []
-        self.result_dict = {}
+        self.__ScheduleNextOutput()
 
-        self.vtk_post_path_directory = os.path.join(param["folder_name"].GetString())
+        # Print background grid model part
         if not self.model_part.ProcessInfo[KratosMultiphysics.IS_RESTARTED]:
-            kratos_utils.DeleteDirectoryIfExisting(self.vtk_post_path_directory)
-            os.makedirs(str(self.vtk_post_path_directory))
+            grid_settings = KratosMultiphysics.Parameters()
+            grid_settings.AddString("model_part_name","Background_Grid")
+            grid_settings.AddString("file_format",settings["file_format"].GetString())
+            grid_settings.AddDouble("output_precision",settings["output_precision"].GetDouble())
+            grid_settings.AddBool("output_sub_model_parts",settings["output_sub_model_parts"].GetBool())
+            grid_settings.AddString("output_path",settings["output_path"].GetString())
+            grid_settings.AddBool("save_output_files_in_folder",settings["save_output_files_in_folder"].GetBool())
+            background_grid = model["Background_Grid"]
+            output_filename = f"Background_Grid_{background_grid.GetCommunicator().MyPID()}"
+            KratosMultiphysics.VtkOutput(background_grid, grid_settings).PrintOutput(output_filename)
```

Comment:
Notice that now, when the process is defined, the background grid is also written using the `VtkOutput` class in the core. I think it is useful since in this way we write the vtk data of the domain where the particles are moving without adding the `vtk_output_process` in the `ProjectParameters.json`